File: Group_Anagram/group_anagram.js
/**
 * @param {string[]} strs - Array of string to find group anagram
 * @returns {string[][]} returns the group anagrams in array
 */
var groupAnagrams = function (strs) {
	const groupAnagrams = new Map();

	for (let i = 0; i < strs.length; i++) {
		let keyStr = strs[i].split("").sort().join("");

		groupAnagrams.has(keyStr)
			? groupAnagrams.get(keyStr).push(strs[i])
			: groupAnagrams.set(keyStr, [strs[i]]);
	}

	const result = [];

	for (const group of groupAnagrams.values()) {
		result.push(group);
	}

	return result;
};

File: Group_Anagram/group_anagram.dart
class Solution {
  List<List<String>> groupAnagrams(List<String> strs) {
    final Map<String, List<String>> groupAnagrams = {};

    for (var str in strs) {
      var keyStr = str.split('')
        ..sort()
        ..join("");

      if (groupAnagrams.containsKey(keyStr)) {
        groupAnagrams[keyStr]!.add(str);
      } else {
        groupAnagrams[keyStr] = [str];
      }
    }

    final List<List<String>> result = [];
    for (var group in groupAnagrams.values) {
      result.add(group);
    }

    return result;
  }
}
